app/services/mixtape_maker.rb

```diff
@@ -0,0 +1,15 @@
+class MixtapeMaker
+
+  def self.generate(no_song = 5, artists:)
+    new.generate(no_song, artists: artists)
+  end
+
+  def generate(no_song = 5, artists:)
+    if artists.blank?
+      Song.limit(no_song).order("RANDOM()")
+    else
+        Song.where(artist: artists[:id]).limit(no_song).order("RANDOM()")
```

Comment:
So you might want something like this:

```
def artists_for_names
  Artist.where(name: artist_names)
end
````

That will do a SQL `where NAME IN ('some name', 'some other name')`

and return a collection of `Artist`s.

Then 

`Song.where(artist: artists).order_by_rand.limit(number_of_songs)`

Will be smart enough to do another SQL "where IN" and just use the ids.